lectures/numpy.md

```diff
@@ -438,6 +438,409 @@ A * B
 (numpy_matrix_multiplication)=
 In particular, `A * B` is *not* the matrix product, it is an element-wise product.
 
+(broadcasting)=
+#### Broadcasting
+
+In element-wise operations above, arrays may not have the same shape.
+ 
+NumPy will automatically expand arrays to the same shape whenever possible.
+
+This useful (but sometimes confusing) feature in NumPy is called **broadcasting**.
+
+For example, `a` is a 3 by 3 2-dimensional array (`a -> (3, 3)`), but `b` is an array with three elements (`b -> (3,)`).
```

Comment:
* For example, *suppose* 
* "but" should be "while" (after comma)
* "3 by 3"  should be $3 \times 3$